### ISSUE TITLE
Run requests in parallel

### ISSUE THREAD

**Issue by dcastro** [serokell/coffer#4]
Most commands need to retrieve _all_ entries descending from either the root directory or some other directory.
When we have hundreds of entries stored in vault, this will mean doing hundreds of requests (1 per directory + 1 per entry).

Unfortunately, Vault's API does not support batch requests.

At the moment, these requests are done sequentially. We'd see a massive speedup if they were run concurrently. 

Commands where this could be applied:

* `view` / `find`: read directories/entries concurrently
* `rename` / `copy`: read / write entries
* `rename` / `delete --recursive`: delete entries

**Comment by sancho20021** [comment]
So what is the final task? To rewrite commands in `Command.hs` using parallel requests? I also didn't understand what @MagicRB meant about dependent types, but if I understood correctly it is not needed, if we parallelize things in `Command.hs` and not on lower level of `readSecrets`.

**Comment by dcastro** [comment]
@sancho20021 yeah let's try solving this at the `Command.hs` level, seems like the simplest most straightforward solution, possibly using `sequenceConcurrently` to parallelize where appropriate. I haven't tried it, but from a cursory glance, looks like it should work.
